Assets/LeapMotion/Core/Scripts/Utils/PostProcessProvider.cs

```diff
@@ -0,0 +1,138 @@
+/******************************************************************************
+ * Copyright (C) Leap Motion, Inc. 2011-2018.                                 *
+ * Leap Motion proprietary and  confidential.                                 *
+ *                                                                            *
+ * Use subject to the terms of the Leap Motion SDK Agreement available at     *
+ * https://developer.leapmotion.com/sdk_agreement, or another agreement       *
+ * between Leap Motion and you, your company or other organization.           *
+ ******************************************************************************/
+
+using UnityEngine;
+using Leap.Unity.Attributes;
+
+namespace Leap.Unity {
+
+  public abstract class PostProcessProvider : LeapProvider {
+
+    [Tooltip("The LeapProvider whose output hand data will be copied, modified, and "
+           + "output by this post-processing provider.")]
+    [SerializeField]
+    [OnEditorChange("inputLeapProvider")]
+    protected LeapProvider _inputLeapProvider;
+    public LeapProvider inputLeapProvider {
+      get { return _inputLeapProvider; }
+      set {
+        validateInput();
```

Comment:
Doesn't make sense to call validateInput here, you want it at the _end_ of the method once _inputLeapProvider has been assigned.

Reply:
will take a look